### PR TITLE
Update to use Aluminum v0.5.0

### PR DESCRIPTION
**Pull request by benson31**
The main significant change is the replacement of "MPICUDABackend" with the more aptly named "HostTransferBackend".

### CODE REVIEW

**Review by bvanessen — CHANGES_REQUESTED** [review]
Shouldn't the Cmake be updated to require a new version of Aluminum?

**Comment by benson31** [comment]
Yeah, or we could do a compatibility typedef based on the Aluminum version, if we care about back-portability at all. Thoughts?

**Comment by bvanessen** [comment]
Let’s use a typedef since we don’t have a v0.1 release yet.

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

**Comment by bvanessen** [comment]
Actually, let’s get the versions instead, with a v0.1 release.

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

**Review by naoyam — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -146,13 +146,14 @@ if (H2_HAS_CUDA OR H2_HAS_HIP_ROCM)
...
-find_package(Aluminum NO_MODULE QUIET
+set(H2_MINIMUM_ALUMINUM_VERSION 0.5.0)
+find_package(Aluminum ${H2_MINIMUM_ALUMINUM_VERSION} NO_MODULE QUIET
   HINTS ${Aluminum_DIR} ${ALUMINUM_DIR} ${AL_DIR}
   $ENV{Aluminum_DIR} $ENV{ALUMINUM_DIR} $ENV{AL_DIR}
   PATH_SUFFIXES lib64/cmake/aluminum lib/cmake/aluminum
   NO_DEFAULT_PATH)
 if (NOT Aluminum_FOUND)
-  find_package(Aluminum NO_MODULE REQUIRED)
+  find_package(Aluminum ${H2_MINIMUM_ALUMINUM_VERSION NO_MODULE REQUIRED)
```
Comment (oyamay) [inline CMakeLists.txt:156]: @benson31 This line lacks a closing bracket.

**Review by oyamay — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (benson31) [reply CMakeLists.txt]: Fixed. Thanks!

**Review by bvanessen — APPROVED** [review]

**Pull request merged**